_integration-tests/tests/confluentinc_confluent-kafka-go.v2/kafka.go

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build integration
+
+package kafka
+
+import (
+	"context"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"orchestrion/integration/utils"
+	"orchestrion/integration/validator/trace"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/redpanda"
+)
+
+var (
+	topic     = "gotest"
+	partition = int32(0)
+)
+
+type TestCase struct {
+	server *redpanda.Container
+	addr   []string
+}
+
+func (tc *TestCase) Setup(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows")
+	}
+	var (
+		err error
+		ctx = context.Background()
+	)
+	tc.server, err = redpanda.Run(ctx,
+		"docker.redpanda.com/redpandadata/redpanda:v24.2.1",
+		redpanda.WithAutoCreateTopics(),
+		testcontainers.WithLogger(testcontainers.TestLogger(t)),
+		utils.WithTestLogConsumer(t),
+	)
+	if err != nil {
+		t.Skipf("Failed to start kafka test container: %v\n", err)
+	}
```

Comment:
```suggestion
		utils.AssertTestContainersError(t, err)
```

this function already exists to check whether the error should mark the test as skipped or not